### ISSUE TITLE
Inflexibility in ignoring whitespace in chains

### ISSUE THREAD

**Issue by ammongit** [ammongit/rshell#27]
The following produce syntax errors even though they shouldn't:
```sh
{ echo hi; } ;
             ^

 {echo hi;}
 ~^

{ {echo hi;}; }
  ~~^

echo abc & echo def &
         ~^
```

This is intimately related with #18. We'll have to think of redo whitespace handling to allow flexibility for groups just like we have them for comands.

**Comment by ammongit** [comment]
Whenever the parser sees `T_WHITESPACE`, it starts looking for a command. I believe this is the source of the shift/reduce conflict found in the mentioned issue above. Perhaps we can create our own replacements for `{` and `}` that are like group: they consume arbitrary instances of `T_WHITESPACE` and simply ignore it.

**Comment by ammongit** [comment]
Fixed in #28
